### DISCUSSION TITLE
react 18 is incompatible with wagmi?

### DISCUSSION THREAD

**Discussion by Iriza-peace** [wevm/wagmi#1629] [Q&A]
I m having this error
react 18 is incompatible with wagmi 
`$ npm install -D vite
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: wagmi@0.2.28
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   peer react@">= 16.8.0" from react-cool-dimensions@2.0.7    
npm WARN   node_modules/react-cool-dimensions
npm WARN     react-cool-dimensions@"^2.0.7" from @thirdweb-dev/react@3.6.9
npm WARN     node_modules/@thirdweb-dev/react
npm WARN   13 more (react-dom, react-router, react-router-dom, ...)   
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer react@"^17.0.0" from wagmi@0.2.28
npm WARN node_modules/wagmi
npm WARN   wagmi@"^0.2.28" from @thirdweb-dev/react@3.6.9
npm WARN   node_modules/@thirdweb-dev/react
npm WARN
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@"^17.0.0" from wagmi@0.2.28
npm WARN   node_modules/wagmi
npm WARN     wagmi@"^0.2.28" from @thirdweb-dev/react@3.6.9
npm WARN     node_modules/@thirdweb-dev/react
npm ERR! code 128
npm ERR! An unknown git error occurred
npm ERR! command git --no-replace-objects ls-remote ssh://git@github.com/Bundlr-Network/avsc.git
npm ERR! git@github.com: Permission denied (publickey).
npm ERR! fatal: Could not read from remote repository.
npm ERR!
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.`

**Comment by tmm** [comment]
wagmi is works with both React 17 and 18. We run our tests in CI [against both](https://github.com/wagmi-dev/wagmi/blob/main/.github/workflows/main.yml#L140).
